shell/platform/android/io/flutter/plugin/common/StandardCodec.java

```diff
@@ -0,0 +1,334 @@
+package io.flutter.plugin.common;
+
+import java.io.ByteArrayOutputStream;
+import java.math.BigInteger;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+
+/**
+ * A {@link MessageCodec} using binary messages in Flutter standard encoding.
```

Comment:
Is there a spec for this protocol somewhere?  If so, can we link to it?  Does it match the Dart SendPort/ReceivePort protocol in a meaningful way?